hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -406,6 +407,9 @@ public void onResponse(Data response) {
                 public void onFailure(Throwable t) {
                 }
             });
+            if (getMapConfig().isStatisticsEnabled()) {
```

Comment:
I don't know if this is a good thing. Now you are going to introduce a new object for every map update and this causes jitter.